experimental/fs.go

```diff
@@ -9,13 +9,16 @@ import (
 )
 
 // WithFS overrides fs.FS in the context-based manner. Caller needs to take
-// responsibility for closing the filesystem.
+// responsibility for closing the returned api.Closer.
 //
 // Note: This has the same effect as the same function on wazero.ModuleConfig.
-func WithFS(ctx context.Context, fs fs.FS) (context.Context, api.Closer) {
+func WithFS(ctx context.Context, fs fs.FS) (context.Context, api.Closer, error) {
```

Comment:
note this breaks the experimental API a bit, though we may in the future see if we can do without this. e.g. if instantiation performance is good enough, it would be faster for everyone to not have to check the Go context prior to any filesystem operation cc @knqyf263